router/base-http/src/main/scala/io/buoyant/router/http/AddForwardedHeader.scala

```diff
@@ -151,35 +174,81 @@ object AddForwardedHeader {
     }
   }
 
-  class LabelingProxy(
-    byLabeler: Labeler.By,
-    forLabeler: Labeler.For,
-    underlying: ServiceFactory[Request, Response]
-  ) extends ServiceFactoryProxy(underlying) {
-
-    override def apply(conn: ClientConnection): Future[Service[Request, Response]] = {
-      // The `by` and `for` labelers are computed once per
-      // connection. This means that a randomized labeler, for
-      // instance, may reuse labels throughout a client's lifetime.
-      val byl = byLabeler(conn.localAddress)
-      val forl = forLabeler(conn.remoteAddress)
-      val filter = new AddForwardedHeader(byl, forl)
-      self.apply(conn).map(filter.andThen(_))
+  object H1 {
+    case class Enabled(enabled: Boolean)
+    implicit object Param extends Stack.Param[Enabled] {
+      // The RFC indicates that this feature should be disabled by default.
+      val default = Enabled(false)
+    }
+
+    class LabelingProxy(
+      byLabeler: Labeler.By,
+      forLabeler: Labeler.For,
+      underlying: ServiceFactory[Request, Response]
+    ) extends ServiceFactoryProxy(underlying) {
+
+      override def apply(conn: ClientConnection): Future[Service[Request, Response]] = {
+        // The `by` and `for` labelers are computed once per
+        // connection. This means that a randomized labeler, for
+        // instance, may reuse labels throughout a client's lifetime.
+        val byl = byLabeler(conn.localAddress)
+        val forl = forLabeler(conn.remoteAddress)
+        val filter = new AddForwardedHeader(byl, forl)
+        self.apply(conn).map(filter.andThen(_))
+      }
     }
+
+    val module: Stackable[ServiceFactory[Request, Response]] =
```

Comment:
The module is protocol specific (and I think this is correct since it should instantiate the appropriate AddForwarededHeader filter to pass to LabelingProxy) so should live in router/http and router/h2